### PR TITLE
Create action file enos.py

### PR DESCRIPTION
**Pull request by amuraleedhar**
##### SUMMARY
Creating action plugin for the modules for enos from Lenovo.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/plugins/action/enos.py

##### ANSIBLE VERSION
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible-2.5.0-py2.7.egg/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]

##### ADDITIONAL INFORMATION

This is an effort to create modules for eos based switches from Lenovo.

### CODE REVIEW

File: lib/ansible/plugins/action/enos_facts.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    def run(self, tmp=None, task_vars=None):
+
+        if self._play_context.connection != 'local':
+            return dict(
+                failed=True,
+                msg='invalid connection specified, expected connection=local, '
+                    'got %s' % self._play_context.connection
+            )
+        provider = self.load_provider()
```
Comment (ganeshrn) [inline lib/ansible/plugins/action/enos_facts.py]: In 2.5 the logic to load provider arguments is moved at a commonplace [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network_common.py#L340).
Please refer action plugin [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/ios.py#L48) for more details.

File: lib/ansible/plugins/action/enos_facts.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        task_vars['ansible_socket'] = socket_path
+
+        if self._play_context.become_method == 'enable':
+            self._play_context.become = False
+            self._play_context.become_method = None
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+        return result
+
+    def load_provider(self):
```
Comment (ganeshrn) [inline lib/ansible/plugins/action/enos_facts.py]: This method not required after above refactor

File: lib/ansible/plugins/action/enos_facts.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            if key != 'provider' and key not in provider:
+                if key in self._task.args:
+                    provider[key] = self._task.args[key]
+                elif 'fallback' in value:
+                    provider[key] = self._fallback(value['fallback'])
+                elif key not in provider:
+                    provider[key] = None
+        return provider
+
+    def _fallback(self, fallback):
```
Comment (ganeshrn) [inline lib/ansible/plugins/action/enos_facts.py]: Same as above

File: lib/ansible/plugins/action/enos_facts.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        # display.vvvv('Again socket_path: %s' % socket_path, pc.remote_addr)
+        if not socket_path:
+            return {'failed': True,
+                    'msg': 'unable to open shell. Please see: ' +
+                           'https://docs.ansible.com/ansible/network_debug_troubleshooting.html#unable-to-open-shell'}
+
+        # make sure we are in the right cli context which should be
+        # enable mode and not config module
+        # rc, out, err = connection.exec_command('prompt()')#'show run'
+        # if str(out).strip().endswith(')#'):
```
Comment (ganeshrn) [inline lib/ansible/plugins/action/enos_facts.py]: Since the code here is commented out can it be removed?

**Review by ganeshrn — COMMENTED** [review]
For the network platform that supports persistent connection, the action handler is invoked based on the prefix of the module name. So if you are planning to add more `enos` modules you add an entry for `enos` [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/config/base.yml#L1296) and rename the action plugin file to `enos.py`  so that same action plugin is invoked for other `enos` modules as well

File: lib/ansible/plugins/action/enos_facts.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+        display.vvv('using connection plugin %s' % pc.connection, pc.remote_addr)
+        connection = self._shared_loader_obj.connection_loader.get('persistent', pc, sys.stdin)
+
+        socket_path = connection.run()
+        display.vvvv('socket_path: %s' % socket_path, pc.remote_addr)
+        # while(socket_path is None and count < 2):
+        #    socket_path = self._shared_loader_obj.connection_loader.get('persistent', pc, sys.stdin).run()
+        #    count = count+1
+        # display.vvvv('Again socket_path: %s' % socket_path, pc.remote_addr)
```
Comment (ganeshrn) [inline lib/ansible/plugins/action/enos_facts.py]: Please remove the commented out code.

**Comment by amuraleedhar** [comment]
Ganesh, Please review again. I tried to incorporate all your review comments

**Review by gundalow — APPROVED** [review]

**Pull request merged**